test/Microsoft.ML.Functional.Tests/Validation.cs

```diff
@@ -49,5 +51,48 @@ void CrossValidation()
             foreach (var result in cvResult)
                 Common.CheckMetrics(result.metrics);
         }
+
+        /// <summary>
+        /// Train with validation set.
+        /// </summary>
+        [Fact]
+        public void TrainWithValidationSet()
+        {
+            var mlContext = new MLContext(seed: 1, conc: 1);
+
+            // Get the dataset
+            var data = mlContext.Data.CreateTextLoader(TestDatasets.housing.GetLoaderColumns(), hasHeader: true)
+                .Read(BaseTestClass.GetDataPath(TestDatasets.housing.trainFilename));
+            (var trainData, var validData) = mlContext.Regression.TrainTestSplit(data, testFraction: 0.2);
+
+            // Create a pipeline to featurize the dataset
+            var pipeline = mlContext.Transforms.Concatenate("Features", new string[] {
+                    "CrimesPerCapita", "PercentResidental", "PercentNonRetail", "CharlesRiver", "NitricOxides", "RoomsPerDwelling",
+                    "PercentPre40s", "EmploymentDistance", "HighwayDistance", "TaxRate", "TeacherRatio"})
+                .Append(mlContext.Transforms.CopyColumns("Label", "MedianHomeValue"))
+                .AppendCacheCheckpoint(mlContext) as IEstimator<ITransformer>;
+
+            // Preprocess the datasets
+            var preprocessor = pipeline.Fit(trainData);
+            var preprocessedTrainData = preprocessor.Transform(trainData);
+            var preprocessedValidData = preprocessor.Transform(validData);
+
+            // Train the model with a validation set
+            var trainedModel = mlContext.Regression.Trainers.FastTree(numTrees: 2)
```

Comment:
>trainedModel [](start = 16, length = 12)

add early stopping criteria in the options so that the validation set is actually useful